AngularApp/src/app/diagnostic-data/components/detector-view/detector-view.component.ts

```diff
@@ -45,11 +46,30 @@ export class DetectorViewComponent implements OnInit {
   @Input() isSystemInvoker: boolean = false;
 
 
-  constructor(@Inject(DIAGNOSTIC_DATA_CONFIG) config: DiagnosticDataConfig, private telemetryService: TelemetryService) {
+  constructor(@Inject(DIAGNOSTIC_DATA_CONFIG) config: DiagnosticDataConfig, private telemetryService: TelemetryService, private detectorControlService: DetectorControlService) {
     this.isPublic = config && config.isPublic;
   }
 
   ngOnInit() {
+    this.loadDetector();
+
+    this.detectorControlService.update.subscribe(validUpdate => {
+      if (validUpdate) {
+
```

Comment:
Is there any operations should be included here? What will happhen if the update is not valid?